src/category_theory/monoidal/internal/equivalence.lean

```diff
@@ -164,6 +164,242 @@ begin
+    ←tensor_comp, category.comp_id, iso.hom_inv_id_app, tensor_id, category.id_comp],
+  simp only [category.assoc, iso.hom_inv_id, category.comp_id],
+  congr,
+end
+
+lemma pentagon' (W X Y Z : D) :
+  tensor_hom' α (associator' α W X Y).hom (𝟙 Z) ≫ (associator' α W (tensor_obj' α X Y) Z).hom ≫
+    tensor_hom' α (𝟙 W) (associator' α X Y Z).hom =
+  (associator' α (tensor_obj' α W X) Y Z).hom ≫ (associator' α W X (tensor_obj' α Y Z)).hom :=
+begin
+  conv_rhs { rw [associator'_hom, associator'_hom, ←α.functor.map_comp] },
+  rw [pentagon'_aux0, ←pentagon],
+  conv_lhs { simp only [associator'_hom, tensor_hom'_def, ←α.functor.map_comp,
+    equivalence.inv_fun_map] },
+  rw [pentagon'_aux1],
+end
+
+lemma triangle'_aux1 (X Y : D) :
+  (α.unit_iso.inv.app _ ⊗ 𝟙 (α.inverse.obj Y)) ≫
+  (α_ (α.inverse.obj X) (α.inverse.obj (tensor_unit' α)) (α.inverse.obj Y)).hom ≫
+    (𝟙 (α.inverse.obj X) ⊗
+       (α.unit_iso.inv.app (𝟙_ C) ⊗ 𝟙 (α.inverse.obj Y)) ≫ (λ_ (α.inverse.obj Y)).hom) =
+  (α.unit_iso.inv.app _ ≫
+    ((tensor_hom (𝟙 (α.inverse.obj _)) (α.unit_iso.inv.app _)) ≫
+    (ρ_ (α.inverse.obj X)).hom)) ⊗ 𝟙 _ :=
+begin
+  simp only [id_tensor_comp, comp_tensor_id, associator_conjugation, category.assoc],
+  rw [←tensor_comp, category.id_comp],
+  congr' 3,
+  erw [←triangle, ←category.assoc (α_ _ _ _).inv, iso.inv_hom_id,
+    category.id_comp, ←tensor_comp, category.id_comp],
+end
+
+lemma triangle'_aux2 (X Y : D) :
+  (𝟙 (α.inverse.obj X) ⊗ α.unit_iso.inv.app (𝟙_ C) ⊗ 𝟙 (α.inverse.obj Y)) ≫
+  (α_ (α.inverse.obj X) ((𝟭 C).obj (𝟙_ C)) (α.inverse.obj Y)).inv ≫
+    ((ρ_ (α.inverse.obj X)).hom ⊗ 𝟙 (α.inverse.obj Y)) =
+  𝟙 _ ⊗ ((tensor_hom (α.unit_iso.inv.app _) (𝟙 _)) ≫ (λ_ _).hom) :=
+begin
+  simp only [id_tensor_comp],
+  erw [←tensor_comp, category.id_comp, ←triangle, ←category.assoc (α_ _ _ _).inv,
+    iso.inv_hom_id, category.id_comp, ←tensor_comp, category.id_comp],
+end
+
+lemma triangle' (X Y : D) :
+  (associator' α X (tensor_unit' α) Y).hom ≫ tensor_hom' α (𝟙 X) (left_unitor' α Y).hom =
+  tensor_hom' α (right_unitor' α X).hom (𝟙 Y) :=
+begin
+  simp only [associator'_hom, functor.map_comp, left_unitor'_hom, iso.app_hom, category.assoc,
+    right_unitor'_hom, tensor_hom', functor.map_id, equivalence.inv_fun_map,
+    equivalence.unit_inverse_comp, category.comp_id, iso.hom_inv_id_app_assoc,
+    id_tensor_comp, comp_tensor_id, associator_conjugation],
+  simp only [←α.functor.map_comp, ←tensor_comp, category.id_comp],
+  simp only [←category.assoc (α.unit_iso.hom.app _), iso.hom_inv_id_app],
+  rw [category.id_comp],
+  rw [triangle'_aux1, triangle'_aux2],
+  congr' 1,
+  simp only [comp_tensor_id, associator_conjugation, category.assoc, id_tensor_comp],
+  erw [←triangle, ←category.assoc (α_ _ _ _).inv, iso.inv_hom_id, category.id_comp, ←tensor_comp,
+    category.id_comp],
+  congr' 1,
+end
+
 end of_equivalence
 
 def of_equivalence : monoidal_category D :=
```

Comment:
Don't we already have this in https://github.com/leanprover-community/mathlib/blob/master/src/category_theory/monoidal/transport.lean ?

Reply:
Indeed! Thanks for the pointer.

Reply:
Correct me if I'm wrong, I think the Monoidal structure for `Module R` only works if the universe level of the ring and module is the same. So to transfer Monoidal structure of `Module Z`, `AddCommGroup` must have universe level zero.

Reply:
[Module.monoidal_category](https://leanprover-community.github.io/mathlib_docs/algebra/category/Module/monoidal.html#Module.monoidal_category) could be generalized to universe level `max u v` (with `v` a free variable); you just need to `ulift.{v}` the `tensor_unit` (which is the ring `R` itself). But the most convenient solution is probably to transport `Module (ulift.{v} ℤ)` rather than `Module ℤ`.